portfolio-project-matching-app/backend/daoProject.js

```diff
@@ -0,0 +1,482 @@
+import {
+    // CREATE
+    addNewDoc,
+    addNewDocWithId,
+    // READ
+    getCollectionSnapshot,
+    getCollectionSnapshotByCriteria,
+    getDocReferenceById,
+    getDocSnapshotById,
+    // UPDATE
+    updateDocument,
+    // DELETE
+    deleteDocById,
+} from '../Firebase/clientApp.ts'
+import { User } from '../models/User'
+import { Project } from '../models/Project'
+import { Technology } from '../models/Technology'
+
+/*
```

Comment:
The majority of these functions will remain unused. I've imported many of the `READ` type functions into `dao.js` along with the `createLike` and `deleteLike` functions. Other than that, they probably won't be used, but we might not want to delete them quite yet - they could come in handy.